src/inspector/node_string.cc

```diff
@@ -1,78 +1,57 @@
 #include "node_string.h"
+#include "crdtp/json.h"
 #include "node/inspector/protocol/Protocol.h"
 #include "simdutf.h"
 #include "util-inl.h"
 
-namespace node {
-namespace inspector {
-namespace protocol {
-namespace StringUtil {
-
-size_t kNotFound = std::string::npos;
+namespace crdtp {
 
-// NOLINTNEXTLINE(runtime/references) V8 API requirement
-void builderAppendQuotedString(StringBuilder& builder,
-                               const std::string_view string) {
-  builder.put('"');
-  if (!string.empty()) {
-    size_t expected_utf16_length =
-        simdutf::utf16_length_from_utf8(string.data(), string.length());
-    MaybeStackBuffer<char16_t> buffer(expected_utf16_length);
-    // simdutf::convert_utf8_to_utf16 returns zero in case of error.
-    size_t utf16_length = simdutf::convert_utf8_to_utf16(
-        string.data(), string.length(), buffer.out());
-    // We have that utf16_length == expected_utf16_length if and only
-    // if the input was a valid UTF-8 string.
-    if (utf16_length != 0) {
-      CHECK_EQ(expected_utf16_length, utf16_length);
-      escapeWideStringForJSON(reinterpret_cast<const uint16_t*>(buffer.out()),
-                              utf16_length,
-                              &builder);
-    }  // Otherwise, we had an invalid UTF-8 input.
+bool ProtocolTypeTraits<std::string>::Deserialize(DeserializerState* state,
+                                                  std::string* value) {
+  if (state->tokenizer()->TokenTag() == cbor::CBORTokenTag::STRING8) {
+    span<uint8_t> cbor_span = state->tokenizer()->GetString8();
+    value->assign(reinterpret_cast<const char*>(cbor_span.data()),
```

Comment:
For extra safety, I think we should check value before usage: 

Dcheck_not_null(value)

Reply:
This is a `crdtp::ProtocolTypeTraits` implementation, like https://github.com/legendecas/node/blob/afcb230cb8dea9c8f6255c3319ba97c0d8ea37cd/deps/inspector_protocol/crdtp/protocol_core.h#L181-L188. It is not intended for direct invocation. It is usually invoked by generated protocol codes. We can definitely add extra check, yet it is not really necessary IMO.

Reply:
I agree. If it's not too much trouble, I think we should add it (just in case) as a dcheck. But I'm leaving this to your decision.